components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,405 @@
+				class="d2l-offscreen"
+				type="button"
+				@click="${this._toggleExpand}"
+				@focus="${this._onFocus}"
+				@blur="${this._onBlur}"
+			>${this.expandCollapseLabel}</button>
+			<div class="d2l-collapsible-panel" @click="${this._handlePanelClick}">
+				<div class="d2l-collapsible-panel-top-sentinel"></div>
+				${this._renderHeader()}
+				<d2l-expand-collapse-content ?expanded="${this.expanded}">
+					<div class="d2l-collapsible-panel-content">
+						<slot name="content"></slot>
+					</div>
+				</d2l-expand-collapse-content>
+				<d2l-expand-collapse-content ?expanded="${!this.expanded}">
+					<div class="d2l-collapsible-panel-summary">
+						<slot name="summary" @slotchange="${this._handleSummarySlotChange}"></slot>
+					</div>
+				</d2l-expand-collapse-content>
+			</div>
+		`;
+	}
+
+	_handleActionsClick(e) {
+		const actions = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-actions');
+		if (e.target !== actions) {
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderClick(e) {
+		if (this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleHeaderSecondaryClick(e) {
+		const header = this.shadowRoot.querySelector('.d2l-collapsible-panel-header-secondary');
+		if (e.target !== header) {
+			e.stopPropagation();
+		}
+	}
+
+	_handlePanelClick(e) {
+		if (!this.expanded) {
+			this._toggleExpand();
+			e.stopPropagation();
+		}
+	}
+
+	_handleSummarySlotChange() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.add('has-summary');
+	}
+
+	_onBlur() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.remove('focused');
+	}
+
+	_onFocus() {
+		const element = this.shadowRoot.querySelector('.d2l-collapsible-panel');
+		element.classList.add('focused');
+	}
+
+	_renderHeader() {
+		return html`
+			<div class="d2l-collapsible-panel-header" @click="${this._handleHeaderClick}">
+				<div class="d2l-collapsible-panel-header-primary">
+					${this._renderHeading()}
+					<div class="d2l-collapsible-panel-header-actions" @click="${this._handleActionsClick}">
+						<slot name="actions" @slotchange="${this._handleActionsSlotChange}"></slot>
+					</div>
+					<div class="d2l-collapsible-panel-opener">
+						<d2l-icon-custom size="tier1">
+							<svg xmlns="http://www.w3.org/2000/svg" width="10" height="18" fill="none" viewBox="0 0 10 18">
+								<path stroke="#494c4e" stroke-linejoin="round" stroke-width="2" d="m9 9-8 8V1l8 8Z"/>
+							</svg>
+						</d2l-icon-custom>
+					</div>
+				</div>
+				<div class="d2l-collapsible-panel-header-secondary" @click="${this._handleHeaderSecondaryClick}">
+					<slot name="header" @slotchange="${this._handleHeaderSlotChange}"></slot>
+				</div>
+			</div>
+			<div class="d2l-collapsible-panel-divider"></div>
+		`;
+	}
+
+	_renderHeading() {
+		const titleClasses = {
+			'd2l-collapsible-panel-title': true,
+			[`d2l-heading-${this.headingStyle}`]: true,
```

Comment:
Why is this inside an array?

Reply:
heh... I'm using a [computed property name](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names) here so I can programatically set `d2l-heading-___`.

If that's not something we can use, I could replace it with something like this:

```js
d2l-heading-1: this.headingStyle === 1,
d2l-heading-2: this.headingStyle === 2,
d2l-heading-3: this.headingStyle === 3,
d2l-heading-4: this.headingStyle === 4,
```

Reply:
James and I chatted outside the review -- this is the correct usage of computed property names so I see no reason to change it. I was just confused as I'd never seen this syntax before. TIL!